changes.d/5890.feat.md

```diff
@@ -0,0 +1,2 @@
+Cylc Lint to warn users that CYLC_VERSION={{CYLC_VERSION}} is deprecated.
+Additional warnings for ROSE_VERSION and FCM_VERSION in the same manner.
```

Comment:
Maybe hone this message so that it's clearer that we're talking about the special case of setting `CYLC_VERSION` from within the workflow config.